zson/formatter.go

```diff
@@ -14,18 +14,19 @@ import (
 )
 
 type Formatter struct {
-	typedefs  map[string]*zed.TypeNamed
-	permanent map[string]*zed.TypeNamed
-	persist   *regexp.Regexp
-	tab       int
-	newline   string
-	builder   strings.Builder
-	stack     []strings.Builder
-	implied   map[zed.Type]bool
-	colors    color.Stack
-}
-
-func NewFormatter(pretty int, persist *regexp.Regexp) *Formatter {
+	typedefs      map[string]*zed.TypeNamed
+	permanent     map[string]*zed.TypeNamed
+	persist       *regexp.Regexp
+	tab           int
+	newline       string
+	builder       strings.Builder
+	stack         []strings.Builder
+	implied       map[zed.Type]bool
+	colors        color.Stack
+	colorDisabled bool
+}
+
+func NewFormatter(pretty int, colorDisabled bool, persist *regexp.Regexp) *Formatter {
```

Comment:
I think this would be clearer with `color bool` instead.

Reply:
Word.

Reply:
The problem with this is that color = true wouldn't necessarily mean color is enabled since it depends when the writer is outputting to a terminal.

Reply:
Ah, right. To make that work we'd need to get rid of pkt/terminal/color.Enabled. Let's do that later.

Reply:
Agreed I think we should get rid of that but yeah, later.